service/controller/resource/tccpf/template/template_main_record_sets.go

```diff
@@ -6,11 +6,31 @@ const TemplateMainRecordSets = `
   GuestNSRecordSet:
     Type: 'AWS::Route53::RecordSet'
     Properties:
-      HostedZoneId: '{{ .RecordSets.ControlPlanePublicHostedZoneID }}'
+      HostedZoneId: '{{ .RecordSets.ControlPlaneHostedZoneID }}'
       Name: '{{ .RecordSets.ClusterID }}.k8s.{{ .RecordSets.BaseDomain }}.'
       Type: 'NS'
       TTL: '300'
-      ResourceRecords: !Split [ ',', '{{ .RecordSets.GuestHostedZoneNameServers }}' ]
+      ResourceRecords: !Split [ ',', '{{ .RecordSets.TenantHostedZoneNameServers }}' ]
+  {{ if ne .RecordSets.ControlPlaneInternalHostedZoneID "" }}
+  TenantAPIServerRecordSet:
+    Type: 'AWS::Route53::RecordSet'
+    Properties:
+      HostedZoneId: '{{ .RecordSets.ControlPlaneInternalHostedZoneID }}'
+      Name: 'api.{{ .RecordSets.ClusterID }}.k8s.{{ .RecordSets.BaseDomain }}.'
+      Type: 'CNAME'
+      TTL: '300'
+      ResourceRecords:
+        - '{{.RecordSets.TenantAPIPublicLoadBalancer }}'
+  TenantIngressRecordSet:
+    Type: 'AWS::Route53::RecordSet'
+    Properties:
+      HostedZoneId: '{{ .RecordSets.ControlPlaneInternalHostedZoneID }}'
+      Name: '*.{{ .RecordSets.ClusterID }}.k8s.{{ .RecordSets.BaseDomain }}.'
+      Type: 'CNAME'
+      TTL: '300'
+      ResourceRecords:
+        - 'ingress.{{ .RecordSets.ClusterID }}.k8s.{{ .RecordSets.BaseDomain }}'
```

Comment:
One idea, we have key functions already for the structure of these domains AFAIK. Would it make sense to use code for structure instead of encoding domain structure in text templates?

Reply:
It is the same way we template it in tccp stack https://github.com/giantswarm/aws-operator/blob/master/service/controller/resource/tccp/template/template_main_record_sets.go#L70-L78 so just replicating